nixos/modules/services/network-filesystems/orangefs/client.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ...} :
+
+with lib;
+
+let
+  cfg = config.services.orangefs.client;
+
+in {
+  ###### interface
+
+  options = {
+    services.orangefs.client = {
+      enable = mkEnableOption "OrangeFS client daemon";
+
+      extraOptions = mkOption {
+        type = types.str;
+        default = "";
+        description = "Extra command line options for pvfs2-client.";
+      };
+
+      fileSystems = mkOption {
+        description = ''
+          The orangefs file systems to be mounted.
+          This option is prefered over using <option>fileSystems</option> directly since
+          the pvfs client service needs to be running for the mounting.
+        '';
+
+        example = [{
+          mountPoint = "/orangefs";
+          target = "tcp://server:3334/orangefs";
+        }];
+
+        type = with types; listOf (submodule ({ ... } : {
+          options = {
+
+            mountPoint = mkOption {
+              type = types.str;
+              default = "/orangefs";
+              description = "Mount point.";
+            };
+
+            options = mkOption {
+              type = with types; listOf str;
+              default = [];
+              description = "Mount options";
+            };
+
+            target = mkOption {
+              type = types.str;
+              default = null;
+              example = "tcp://server:3334/orangefs";
+              description = "Target URL";
+            };
+          };
+        }));
+      };
+    };
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.orangefs ];
+
+    boot.supportedFilesystems = [ "pvfs2" ];
+    boot.kernelModules = [ "orangefs" ];
+
+    systemd.services.orangefs-client = {
+      requires = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+
+      # Let service settle after network is online
+      # and before mounts are done.
+      # Otherwise client daemon may hang.
+      preStart = "sleep 1";
+      postStart = "sleep 1";
```

Comment:
This looks hacky. Is this what upstream suggests?

Reply:
This is what fedora uses: https://src.fedoraproject.org/rpms/orangefs/blob/master/f/orangefs-client.service

Reply:
I had problems in a virtualbox which looked like a race condition between mount and the client daemon. The way fedora does it is not the recommended way by upstream but it seems to work.

Reply:
Do you know a different way, readiness could be checked instead?

Reply:
Calling the client core directly may solve the problem and remove the internal startup delay.

The service creates a unix datagram socket. Is there a way to track that with systemd?